### PR TITLE
Update Macintosh package libraries

### LINKED ISSUE #20127: Upgrade moose-mpich / moose-tools use latest available libraries
## Reason
Peacock currently does not work with our Conda packages for Macintosh. This is due to at-the-time unavailable libraries built for the necessary version of Clang we now require to use.

## Design
Bump all libraries in use to later versions supporting the new Clang compiler stack.

## Impact
Have Peacock function again, for modern versions of Macintosh.

### PR DESCRIPTION
**Pull request by milljm**
Update all packages involved with the previous inability to run Peacock.

Closes #20127

### CODE REVIEW

File: conda/mpich/conda_build_config.yaml [hunk 1]
```diff
@@ -203,11 +225,11 @@ moose_sysroot:                                              # [linux]
   - sysroot_linux-64 2.17 h4a8ded7_13                       # [linux]
 
 moose_libcxx:                                               # [osx]
-  - libcxx 12.0.1 habf9029_0                                # [not arm64 and osx]
+  - libcxx 12.0.1 habf9029_1                                # [not arm64 and osx]
   - libcxx 12.0.1 h168391b_0                                # [arm64]
 
 moose_libclang:                                             # [osx]
-  - libclang 12.0.1 default_he082bbe_4                      # [not arm64 and osx]
+  - libclang 13.0.0 default_he082bbe_0                      # [not arm64 and osx]
```
Comment (milljm) [inline conda/mpich/conda_build_config.yaml:232]: I know this is strange, but this is what pyqt/qt _demands_. And it seems to not conflict with our 12.x requirements.

File: conda/libmesh-vtk/build.sh [hunk 2]
```diff
@@ -30,7 +30,8 @@ cmake .. -G "Ninja" \
     -DVTK_GROUP_ENABLE_Web:STRING=NO \
     -DCMAKE_OSX_SYSROOT=${CONDA_BUILD_SYSROOT}
 
-ninja install -v
+CORES=$(echo "${CPU_COUNT:-2} / 2" | bc)
+ninja install -v -j $CORES
```
Comment (milljm) [inline conda/libmesh-vtk/build.sh:34]: I found while building on my machine, it is destroyed by oversubscribing. I think Conda treats threads as cores.

**Comment by cticenhour** [comment]
I'm seeing several instances in `moose-tools` where we have earlier versions (say `vtk`, `numpy`, etc.) for Python 3.6 and 3.7 versus the restored versions (3.8 and 3.9). Any chance these can be updated so that we're using consistent versions?

File: conda/mpich/conda_build_config.yaml (hunk 1 above)
Comment (cticenhour) [reply conda/mpich/conda_build_config.yaml:232]: I seem to remember always having to update `libclang` with the main clang version we use in the package. I find it interesting that you didn't need to adjust `libcxx` as well.

**Comment by milljm** [comment]
> I'm seeing several instances in `moose-tools` where we have earlier versions for certain packages (say `vtk`, `numpy`, etc.) for Python 3.6 and 3.7 versus those packages corresponding to 3.8 and 3.9. Any chance these can be updated so that we're using consistent versions?

Indeed, working on those now. I see an updated PyQt for Python 3.7. So that is a _good_ sign!

**Comment by cticenhour** [comment]
Oops, I jumped the gun on my comment - good to hear!

File: conda/libmesh-vtk/build.sh (hunk 2 above)
Comment (cticenhour) [reply conda/libmesh-vtk/build.sh:34]: In favor of this, as conda build has definitely used more than I thought it should in the past. 👍🏻

File: conda/mpich/conda_build_config.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 ## Any changes made will require additional changes to any item above that.
 
 moose_libmesh:
-  - moose-libmesh 2022.01.19 build_1
+  - moose-libmesh 2021.12.15 build_2
```
Comment (loganharbour) [inline conda/mpich/conda_build_config.yaml:5]:
```suggestion
  - moose-libmesh 2022.01.19 build_2
```

**Review by loganharbour — DISMISSED** [review]

**Comment by milljm** [comment]
While this may pass... we're seeing an interesting occurrence of hung processes. We need to investigate further.

**Comment by cticenhour** [comment]
Note to ourselves later - hanging test harness in modules combined runs seems to be related to python: 
```
/Users/user/miniconda3/envs/test_clang13/bin/python -c from multiprocessing.resource_tracker import main;main(6)
```
This system hang seems to occur only when large numbers of tests are being run at once (like "modules combined") instead of when modules or framework are tested individually.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**